### DISCUSSION TITLE
error with handleEditEntry function

### DISCUSSION THREAD

**Discussion by donacrisante** [neuefische/questions#695] [Web] [answered]
Hello everyone,
I've been trying to get my handleEditEntry function working since this morning, but I keep getting the error message that id is not defined. I tried to pass "id" as prop in App.js, EntryList.js and edit.js, but it doesn't work either. I have also rewritten the function several times, but it still doesn't work. Can someone help me with this please? Thank you in advance!

Here is my PR:
https://github.com/donacrisante/Capstone-project/pull/11

**Comment by F-Kirchhoff** [comment, marked as answer]
Hey,
right now I don't see where you actually update your "entries" array such that the updatedEntry is replacing the old entry in that list. At some point you need to take your updatedEntry and somehow replace the old entry in your entries list with this updated one. 
Your "handleSaveEditEntry" function does nothing right now. Maybe the best would be to:
1. reset your edit logic
2. create a function "handleEditEntry" inside your "_app.js"
3. pass this function all the way into your editForm
4. use this function inside the submit callback function and pass into it the updated entry
5. write the logic for replacing the old entry in the entries list inside "handleEditEntry" in _app.js
6. hint: you need to use some kind of `const updatedEntries = entries.map(???)` and `setEntries(updatedEntries)`

Good Luck!

**Comment by donacrisante** [comment]
Hi Felix,
thanks a lot for your feedback and sorry for my late reply. I have restructured my app a little bit and all worked fine after that. Now a have other things to fix.
